src/Settings.js

```diff
@@ -89,37 +108,47 @@ export class Settings {
      *      availableInputTypes: AvailableInputTypes,
      *      featureToggles: AutofillFeatureToggles
      * }>}
-     * @param {AvailableInputTypes} [availableInputTypesOverrides] a migration aid so that macOS can provide data in its old way initially
      */
-    async refresh (availableInputTypesOverrides) {
+    async refresh () {
         this.setFeatureToggles(await this.getFeatureToggles())
-        const availableInputTypesFromRemote = await this.getAvailableInputTypes()
+        this.setAvailableInputTypes(await this.getInitialAvailableInputTypes())
 
-        /** @type {AvailableInputTypes} */
-        const availableInputTypes = {
-            email: false, // not supported yet
-            ...availableInputTypesFromRemote,
-            ...availableInputTypesOverrides
+        return {
+            featureToggles: this.featureToggles,
+            availableInputTypes: this.availableInputTypes
         }
+    }
 
-        // Update the availableInputTypes to take into account the feature toggles.
-        if (!this.featureToggles.inputType_credentials) {
-            availableInputTypes.credentials = false
+    /**
+     * Checks if a mainType/subtype pair can be autofilled with the data we have
+     * @param {SupportedMainTypes} mainType
+     * @param {string} subtype
+     * @returns {Promise<boolean>}
+     */
+    async canAutofillType (mainType, subtype) {
```

Comment:
This new method is used to check if an input type can be autofilled. Thanks to this we no longer access `availableInputTypes` directly. This also allows us to check single input types rather than just the `mainType` like we did before. This allows us to avoid subtle bugs where for example we prompted autofill for credentials on the `username` even though we only have the `password`.